framework/src/utils/MeshBaseDiagnosticsUtils.C

```diff
@@ -66,4 +66,107 @@ checkNonConformalMesh(const std::unique_ptr<MeshBase> & mesh,
   }
   pl->unset_close_to_point_tol();
 }
+
+bool
+checkEdgeOverlap(const std::unique_ptr<Elem> & edge1,
+                 const std::unique_ptr<Elem> & edge2,
+                 const ConsoleStream & console,
```

Comment:
I would remove the console here and instead return a reference to the intersection point

then you can have a good message in the MG with all the info you could want (element concerned, etc)